caikit_nlp/modules/token_classification/filtered_span_classification.py

```diff
@@ -99,18 +99,24 @@ def __init__(
         error.type_check_all(
             "<NLP71653678E>", str, allow_none=True, labels_to_output=labels_to_output
         )
-        classification_task = classifier.TASK_CLASS
+        classification_tasks = classifier._TASK_CLASSES
         error.value_check(
             "<NLP41319814E>",
-            classification_task in ALLOWED_TASKS,
+            any([i for i in classification_tasks if i in ALLOWED_TASKS]),
             f"classifier does not implement one of required tasks: {ALLOWED_TASKS}",
         )
+        classification_task_set = set(classification_tasks)
+        error.value_check(
+            "<NLP41319815E>",
+            len(classification_task_set) == 1,
```

Comment:
From my understanding the goal here is to check if the classifier implements one but not more than one of the ALLOWED_TASKS - can we just check that both (in this case) are not in the ALLOWED_TASKS?

I think this would disallow a case where a multitask classifier may implement, say TextClassification and a theoretical say, SentimentClassification task (that would have no effect on the particular module here)? (might be good to have a small test addition)

Reply:
I'm a bit confused - by "unnecessary" did you mean to remove it?

I'm thinking of the case where [attributes replaced with strings for simplicity]
```
classification_tasks = ['sentiment', 'textclassification'] # from type(classifier).tasks but for simplicity
allowed_tasks = ['textclassification', 'tokenclassification']
```
the `any([i for i in classification_tasks....` should return True as expected but the
```
len(set(classification_tasks)) == 1
```
will return False for the classifier. If the "sentiment" task is perhaps just a more specific implementation, for example, for the text classification task, it probably should not trigger an error and be disallowed from use in this module?